valohai_yaml/objs/parameter_widget.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class ParameterWidget(Item):
+    """An UI widget for editing parameter values."""
+
+    def __init__(
+        self,
+        *,
+        type: str,
+        settings: Optional[dict] = None,
+    ) -> None:
+        self.type = str(type).lower()
+        self.settings = dict(settings or {})
+
+    def serialize(self) -> Optional[Union[SerializedDict, str]]:
+        if self.settings:
+            return {
+                'type': self.type,
+                'settings': self.settings,
+            }
+        return self.type
+
+    @classmethod
+    def parse(cls, data: Optional[Union['ParameterWidget', dict, str]]) -> 'ParameterWidget':
```

Comment:
Based on the code (I swear it had `if data is None` in an earlier revision?!) `Optional` is no longer true here; this would do the wrong thing for `None`.

Either get rid of that –

```suggestion
    def parse(cls, data: Union['ParameterWidget', dict, str]) -> 'ParameterWidget':
```

or alternately make it return `None` for falsy values...

```suggestion
    def parse(cls, data: Optional[Union['ParameterWidget', dict, str]]) -> Optional['ParameterWidget']:
        if not data:
            return None
```

I'd go with option 2, because it would simplify `Parameter.parse` too; you wouldn't need to check whether `widget` is in `data`...

Reply:
I was going for option 2 earlier, but the mypy error made me think its not cool (?)

```
Return type "Optional[ParameterWidget]" of "parse" incompatible with return type "ParameterWidget" in supertype "Item"  [override]
```

I've pushed option 1 as separate commit now.

Reply:
Alright! I'd forgotten we had strictly specified `parse` needs to return the thing.